src/chatgpt-operation/api.js

```diff
@@ -1,50 +1,54 @@
-import { Configuration, OpenAIApi } from "openai";
-
+const OpenAI = require("openai");
 import { defineOperationApi } from "@directus/extensions-sdk";
 import { getSetting } from "../lib/util";
-import { openAIField } from "../configuration/fields";
+import { openAIField, gptModelField } from "../configuration/fields";
 
 export default defineOperationApi({
-	id: "chatgpt-operation",
-	handler: async (
-		{
-			messages,
-			api_key,
-			temperature = 0.5,
-			max_tokens = null,
-			top_p = 1,
-			frequency_penalty = 0,
-			presence_penalty = 0,
-		},
-		{ services, database, getSchema }
-	) => {
-		const { SettingsService } = services;
-		const schema = await getSchema();
-		const settings = new SettingsService({ schema, knex: database });
+    id: "chatgpt-operation",
+    handler: async ({
+        messages,
+        api_key,
+        temperature = 0.5,
+        max_tokens = null,
+        top_p = 1,
+        frequency_penalty = 0,
+        presence_penalty = 0,
+    }, { services, database, getSchema }) => {
+        const { SettingsService } = services;
+        const schema = await getSchema();
+        const settings = new SettingsService({ schema, knex: database });
+
+        const apiKey = await getSetting(settings, openAIField.field, api_key);
+        
+        // Get model directly from database
+        const dbSettings = await database
+            .select(gptModelField.field)
+            .from('directus_settings')
+            .first();
```

Comment:
Can we make use of the `getSetting` function here to pull values from the database? same as the API keys

Reply:
@br41nslug maybe you can help me figure this one out. When I tried to see the getSetting:

```
 "debug": {
      "modelFieldName": "Default_GPT_Model",
      "modelFromGetSetting": "gpt-3.5-turbo",
      "modelFromDB": {
        "Default_GPT_Model": "gpt-4-turbo-preview"
      }
```

For some reason the getSetting was returning my fallback and not the value from the dropdown itself. 

I could not figure this one out besides directly calling the db.

Reply:
`getSetting` doesnt have a `fallback` but rather an `override` which is used to give priority to the current operation config preventing unneeded db calls, you can do defaulting like you have a couple lines below this ```modelFieldSetting || "gpt-4o"```;

Reply:
taking a look now

Reply:
`const model = await getSetting(settings, gptModelField.field, null) || "gpt-4o";`

this is working now. updated!